### PR TITLE
Port the world model lifecycle to lifecycle nodes

### PR DESCRIPTION
**Pull request by mjeronimo**
## Description
* Port the world model to lifecycle nodes

### CODE REVIEW

File: nav2_world_model/src/world_model.cpp [hunk 1]
```diff
@@ -12,67 +12,124 @@
...
-WorldModel::WorldModel(rclcpp::executor::Executor & executor, const string & name)
-: Node(name),
-  tfBuffer_(get_clock()),
-  tfListener_(tfBuffer_)
+WorldModel::WorldModel()
+: nav2_lifecycle::LifecycleNode("world_model")
+{
+  RCLCPP_INFO(get_logger(), "Creating");
+
+  costmap_ros_ = std::make_shared<nav2_costmap_2d::Costmap2DROS>("world_model_global_costmap");
```
Comment (bpwilcox) [inline nav2_world_model/src/world_model.cpp]: I assume (now looking at #664) that the `costmap_2d_ros` creates its own tf_buffer for transforms

Comment (mjeronimo) [reply nav2_world_model/src/world_model.cpp]: @bpwilcox Yes, that's correct.

File: nav2_world_model/src/world_model.cpp [hunk 2]
```diff
@@ -12,67 +12,124 @@
...
+  RCLCPP_INFO(get_logger(), "Cleaning up");
+
+  costmap_ros_->on_cleanup(state);
+  costmap_service_.reset();
+
+  return nav2_lifecycle::CallbackReturn::SUCCESS;
+}
+
+nav2_lifecycle::CallbackReturn
+WorldModel::on_error(const rclcpp_lifecycle::State &)
```
Comment (mhpanah) [inline nav2_world_model/src/world_model.cpp]: should we add `/*state*/` to be consistent with other PR's?

File: nav2_world_model/src/world_model.cpp [hunk 3]
```diff
@@ -12,67 +12,124 @@
...
+{
+  RCLCPP_ERROR(get_logger(), "Handling error state");
+  return nav2_lifecycle::CallbackReturn::SUCCESS;
+}
+
+nav2_lifecycle::CallbackReturn
+WorldModel::on_shutdown(const rclcpp_lifecycle::State &)
```
Comment (mhpanah) [inline nav2_world_model/src/world_model.cpp]: same as above.

File: nav2_world_model/src/world_model.cpp (hunk 2 above)
Comment (mjeronimo) [reply nav2_world_model/src/world_model.cpp]: Yes, will do.

File: nav2_world_model/src/world_model.cpp (hunk 3 above)
Comment (mjeronimo) [reply nav2_world_model/src/world_model.cpp]: Ok.

**Pull request merged**